core/lib/server/pub.ts

```diff
@@ -1275,26 +1261,27 @@ export async function getPubsWithRelatedValuesAndChildren<
 										])
 									)
 								)
-								.leftJoin("pub_values", "pubs.id", "pub_values.pubId")
-								.leftJoin("pub_fields", "pub_fields.id", "pub_values.fieldId")
 								.leftJoin("PubsInStages", "pubs.id", "PubsInStages.pubId")
+								.$if(Boolean(withRelatedPubs), (qb) =>
+									qb
+										.leftJoin("pub_values", (join) =>
+											join.on((eb) =>
+												eb.and([
+													eb("pub_values.pubId", "=", eb.ref("pubs.id")),
+													eb("pub_values.relatedPubId", "is not", null),
+												])
+											)
+										)
+										.select("pub_values.relatedPubId")
```

Comment:
this conditional select is a bit sneaky, but i wanted to be able to avoid joining on the relatedPubIds if it's not necessary.

it's sneaky, bc it's kind of easy to forget, and if you don't use the same condition to select it below `selectFrom(pub_tree as pt)...` you won't get a type error but postgres will yell at you for selecting a column that doesn't exist